### PR TITLE
[Port] SPACEPODS!

### PR DESCRIPTION
**Pull request by SynnGraffkin**
## About The Pull Request
Thanks @kevinz000 for helping with Projectile code.

Ports [Spacepods](https://github.com/yogstation13/Yogstation/pull/4003)  Credit to monster860, steamp0rt

Bugfixes 
[makes turrets shoot spacepods #4083](https://github.com/yogstation13/Yogstation/pull/4083)
[fucking nich broke my shit #7188](https://github.com/yogstation13/Yogstation/pull/7188)
[Fixes bug where you can accidentally destroy your spacepod instead of fixing it #7427](https://github.com/yogstation13/Yogstation/pull/7427)
[fixes spacepods not being able to be examined #7346](https://github.com/yogstation13/Yogstation/pull/7346)

### Tested and works locally

Like paradise's but:

-    flying is more realistic
-    shooting is more realistic
-    you build it using metal rods and protolathe instead of a snowflakey pod fab
-    you can find the parts in maint sometimes
-    flying is more realistic
-    you can actually deconstruct it

Here's the more realistic flying: https://streamable.com/sjotn

and the more realistic shooting: https://streamable.com/k33m0

some code stolen borrowed from this pr: https://github.com/tgstation/tgstation/pull/38169/files

and now here's the mapped in podbay in escape (spacepod not included):

image

## Changelog
:cl:
add: Spacepods! Build it using metal rods and some parts, and you can find the parts you need either in maint (if you're lucky) or build them using a protolathe
fix: Turrets now shoot spacepods
fix: Running out of welding fuel while fixing your spacepod will no longer stack a billion hits on your spacepod
fix: Fixes not being able to examine spacepods
/:cl:

### CODE REVIEW

**Comment by SynnGraffkin** [comment]
waspstation/code/_onclick/adjacent.dm, line 35, column 13:
error: bad keyword argument "target" to /atom/proc/Adjacent
waspstation/code/_onclick/adjacent.dm, line 35, column 13:
error: bad keyword argument "mover" to /atom/proc/Adjacent

Probably won't be an issue since this ran for Yog without Trabis? Time to test locally.

**Comment by SynnGraffkin** [comment]
Works locally, spacepods are a little frail, but it works.

File: waspstation/code/modules/spacepods/spacepod.dm [hunk 1]
```diff
@@ -0,0 +1,688 @@
...
+		return ..()
+	var/list/items = list(cell, internal_tank)
+	items += equipment
+	var/list/item_map = list()
+	var/list/used_key_list = list()
+	for(var/obj/I in items)
+		item_map[avoid_assoc_duplicate_keys(I.name, used_key_list)] = I
+	var/selection = input(user, "Remove which equipment?", null, null) as null|anything in item_map
+	var/obj/O = item_map[selection]
+	if(O && istype(O) && O in contents)
```
Comment (MarkSuckerberg) [inline waspstation/code/modules/spacepods/spacepod.dm:215]:
```suggestion
	if(O && istype(O) && (O in contents))
```

File: waspstation/code/modules/spacepods/spacepod.dm [hunk 2]
```diff
@@ -0,0 +1,688 @@
...
+			add_overlay(image(icon = 'waspstation/icons/obj/spacepods/2x2.dmi', icon_state = "rcs_left", dir = cdir))
+		if(right_thrust)
+			add_overlay(image(icon = 'waspstation/icons/obj/spacepods/2x2.dmi', icon_state = "rcs_right", dir = cdir))
+	if(back_thrust)
+		var/image/I = image(icon = 'waspstation/icons/obj/spacepods/2x2.dmi', icon_state = "thrust")
+		I.transform = matrix(1, 0, 0, 0, 1, -32)
+		add_overlay(I)
+
+/obj/spacepod/MouseDrop_T(atom/movable/A, mob/living/user)
+	if(user == pilot || user in passengers || construction_state != SPACEPOD_ARMOR_WELDED)
```
Comment (MarkSuckerberg) [inline waspstation/code/modules/spacepods/spacepod.dm:468]:
```suggestion
	if(user == pilot || (user in passengers) || construction_state != SPACEPOD_ARMOR_WELDED)
```

File: waspstation/code/modules/spacepods/equipment.dm [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+		cargo_map[avoid_assoc_duplicate_keys("[E.name] ([E.storage.name])", used_key_list)] = E
+	var/selection = input(usr, "Unload which cargo?", null, null) as null|anything in cargo_map
+	var/obj/item/spacepod_equipment/cargo/large/E = cargo_map[selection]
+	if(!selection || !verb_check() || !E || !(E in equipment) || !E.storage)
+		return
+	E.storage.forceMove(loc)
+	E.storage = null
+
+/obj/item/spacepod_equipment/cargo/large/proc/spacepod_mousedrop(obj/spacepod/SP, obj/A, mob/user)
+	if(user == SP.pilot || user in SP.passengers)
```
Comment (MarkSuckerberg) [inline waspstation/code/modules/spacepods/equipment.dm:126]:
```suggestion
	if(user == SP.pilot || (user in SP.passengers))
```

File: waspstation/code/_onclick/adjacent.dm [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+/atom/movable/Adjacent(var/atom/neighbor)
+	if(neighbor == loc)
+		return TRUE
+	if(!isturf(loc))
+		return FALSE
+	if((islist(locs) && locs.len > 1) && (bound_width != world.icon_size || bound_height != world.icon_size))
+		for(var/turf/T in locs) //this is to handle multi tile objects
+			if(T.Adjacent(neighbor, src, src))
+				return TRUE
+	else if(loc.Adjacent(neighbor,target = neighbor, mover = src))
```
Comment (MarkSuckerberg) [inline waspstation/code/_onclick/adjacent.dm:35]:
```suggestion
	var/turf/T = loc
	else if(T.Adjacent(neighbor,target = neighbor, mover = src))
```

File: waspstation/code/modules/research/techweb/all_nodes.dm [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+	design_ids = list("podcargo_seat")
+
+/datum/techweb_node/spacepod_storage
+	id = "spacepod_storage"
+	display_name = "Spacepod Storage"
+	description = "For storing the stuff you find in the far reaches of space"
+	research_costs = list(TECHWEB_POINT_TYPE_GENERIC = 4500)
+	export_price = 4500
+	prereq_ids = list("spacepod_pseat", "high_efficiency")
+	design_ids = list("podcargo_lootbox", "podcargo_crate", "podcargo_ore")
```
Comment (MarkSuckerberg) [inline waspstation/code/modules/research/techweb/all_nodes.dm:80]:
```suggestion
	design_ids = list("podcargo_crate", "podcargo_ore")
```

File: waspstation/code/_onclick/adjacent.dm [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+	if(!isturf(loc))
+		return FALSE
+	if((islist(locs) && locs.len > 1) && (bound_width != world.icon_size || bound_height != world.icon_size))
+		for(var/turf/T in locs) //this is to handle multi tile objects
+			if(T.Adjacent(neighbor, src, src))
+				return TRUE
+	var/turf/T = loc
+	else if(T.Adjacent(neighbor,target = neighbor, mover = src))
+		return TRUE
+	return FALSE
```
Comment (MarkSuckerberg) [inline waspstation/code/_onclick/adjacent.dm:38]:
```suggestion
/atom/movable/Adjacent(var/atom/neighbor)
	var/turf/T = loc
	if(neighbor == loc)
		return TRUE
	if(!isturf(loc))
		return FALSE
	if((islist(locs) && locs.len > 1) && (bound_width != world.icon_size || bound_height != world.icon_size))
		for(var/turf/T in locs) //this is to handle multi tile objects
			if(T.Adjacent(neighbor, src, src))
				return TRUE
	else if(T.Adjacent(neighbor,target = neighbor, mover = src))
		return TRUE
	return FALSE
```
woops

**Comment by SynnGraffkin** [comment]
>wakes up from coma 
>sees Marg update my PR 
>goes back to sleep knowing it may work and I didn't have to do nuffin

**Comment by dinnerwoerror** [comment]
why does the silver surfer look like the antagonist from Deus ex

**Pull request merged**